vagen/envs/tos/tos_env.py

```
@@ -0,0 +1,216 @@
+    """
+    Spatial Gym Environment (exploration only).
+    Adapts ToS to VAGEN's GymImageEnv interface.
+    """
+    def __init__(self, env_config: Dict[str, Any]):
+        super().__init__(env_config)
+        if isinstance(env_config, dict):
+            try:
+                valid_fields = SpatialGymConfig.__dataclass_fields__.keys()
+                filtered_config = {k: v for k, v in env_config.items() if k in valid_fields}
+                self.config = SpatialGymConfig(**filtered_config)
+            except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)